packages/block-library/src/navigation-link/edit.js

```diff
@@ -370,7 +367,9 @@ export default function NavigationLinkEdit( {
 	);
 
 	if ( ! url || isInvalid || isDraft ) {
-		blockProps.onClick = () => setIsLinkOpen( true );
+		blockProps.onClick = () => {
+			setIsLinkOpen( true );
+		};
```

Comment:
onClick handles Enter keypresses, so the onKeydown Enter keypress check is unnecessary.